github/test_github.py

```diff
@@ -0,0 +1,296 @@
+        '''
+        # Mock the commit message
+        mock.return_value = (
+            'This is a test commit\n\n'
+            'Updates: #1234')
+
+        # Parse the commit message
+        commit_msg = handle_github.get_commit_message()
+        commit = handle_github.CommitHandler('glusterfs')
+        issues = commit.parse_commit_message(commit_msg)
+        self.assertListEqual(issues, ['1234'])
+
+        commit.revision_number = 1
+        deduped = commit.remove_duplicates(issues)
+        self.assertListEqual(deduped, issues)
+
+    @patch('handle_github.CommitHandler.get_commit_message_from_gerrit')
+    @patch('handle_github.get_commit_message')
+    def test_second_revision_same_issue(self, mock1, mock2):
+        '''
+        A test to check deduplication when revision number is 2 with same issue
+        in commit message
+        '''
+        # Mock the commit message
+        mock1.return_value = (
+            'This is a test commit\n\n'
+            'Updates: #1234')
+        # Parse the commit message
+        commit_msg = handle_github.get_commit_message()
+        commit = handle_github.CommitHandler('glusterfs')
+        issues = commit.parse_commit_message(commit_msg)
+        self.assertListEqual(issues, ['1234'])
+
+        # Mock the commit message from Gerrit
+        mock2.return_value = (
+            'This is the previous commit\n\n'
+            'Updates: #1234')
+        commit.revision_number = 2
+        deduped = commit.remove_duplicates(issues)
+        self.assertListEqual(deduped, [])
+
+    @patch('handle_github.CommitHandler.get_commit_message_from_gerrit')
+    @patch('handle_github.get_commit_message')
+    def test_second_revision_different_issue(self, mock1, mock2):
+        '''
+        A test to check deduplication when second revision has a different
+        issue from the first revision
+        '''
+        mock1.return_value = (
+            'This is a test commit\n\n'
+            'Updates: #4567')
+        # Parse the commit message
+        commit_msg = handle_github.get_commit_message()
+        commit = handle_github.CommitHandler('glusterfs')
+        issues = commit.parse_commit_message(commit_msg)
+        self.assertListEqual(issues, ['4567'])
+
+        # Mock the commit message from Gerrit
+        mock2.return_value = (
+            'This is the previous commit\n\n'
+            'Fixes: #1234')
+        commit.revision_number = 2
+        deduped = commit.remove_duplicates(issues)
+        self.assertListEqual(deduped, ['4567'])
+
+    @patch('handle_github.CommitHandler.get_commit_message_from_gerrit')
+    @patch('handle_github.get_commit_message')
+    def test_second_revision_new_issue(self, mock1, mock2):
+        '''
+        A test to check deduplication when the secon revision has an added
+        issue
+        '''
+        mock1.return_value = (
+            'This is a test commit\n\n'
+            'Updates: #1234'
+            'Fixes: #4567')
```

Comment:
Isn't this OK, if the issue numbers are different? Or should we keep this as one issue per patch.

Reply:
Handling that use-case just in case it pops up. Do you want us to throw an error instead?